js/coinsbit.js

```diff
@@ -0,0 +1,682 @@
+                    // '1002': InvalidOrder, // {"success":false,"code":1002,"message":"Contract not allow place order!"}
+                    // '10072': AuthenticationError, // Invalid access key
+                    // '10073': AuthenticationError, // Invalid request time
+                    // '10216': InvalidAddress, // {"code":10216,"msg":"No available deposit address"}
+                    // '10232': BadSymbol, // {"code":10232,"msg":"The currency not exist"}
+                    // '30000': BadSymbol, // Trading is suspended for the requested symbol
+                    // '30001': InvalidOrder, // Current trading type (bid or ask) is not allowed
+                    // '30002': InvalidOrder, // Invalid trading amount, smaller than the symbol minimum trading amount
+                    // '30003': InvalidOrder, // Invalid trading amount, greater than the symbol maximum trading amount
+                    // '30004': InsufficientFunds, // Insufficient balance
+                    // '30005': InvalidOrder, // Oversell error
+                    // '30010': InvalidOrder, // Price out of allowed range
+                    // '30016': BadSymbol, // Market is closed
+                    // '30019': InvalidOrder, // Orders count over limit for batch processing
+                    // '30020': BadSymbol, // Restricted symbol, API access is not allowed for the time being
+                    // '30021': BadSymbol, // Invalid symbol
+                    // '33333': BadSymbol, // {"code":33333,"msg":"currency can not be null"}
+                },
+                'broad': {
+                },
+            },
+        });
+    }
+
+    checkExchangeType (methodName, params) { // This method can be moved into base
+        // My proposal is to use separate exchange-type (spot|margin|futures) for each API call. So, we can assume a default to be always 'spot' but typically, in every function we should have created a separate parameter, named 'exchangeType', instead of generic 'type', because 'type' can be a very specific to the individual method, and it's better that there was a specific variable name dedicated for 'exchange-type' meaning. Imho, that shouldn't be 'type' but 'exchangeType'(or alike).
+        const defaultType = this.safeString2 (this.options, methodName, 'defaultType', 'spot');
+        const exchangeType = this.safeString (params, 'exchangeType', defaultType);
+        const supportedExchangeTypes = Object.keys (this.api);
+        if (!supportedExchangeTypes.includes (exchangeType)) {
+            throw new ExchangeError (this.id + " does not support '" + exchangeType + "' type, set exchange.options['defaultType'] to desired: " + supportedExchangeTypes.join (' | ')); // eslint-disable-line quotes
+        }
+        params = this.omit (params, 'exchangeType');
+        return [exchangeType, params];
+    }
+
+    safeMarketObject (element) { // This method can be moved into base
```

Comment:
I would suggest to keep this change outside of the scope of the Coinsbit integration and move these fields inside fetchMarkets. We will address this aspect a bit later when we have a better-unified market structure across all libs.